### ISSUE TITLE
Negative loss

### ISSUE THREAD

**Issue by fuzzyBatman** [qubvel/segmentation_models#95]
What might I be doing wrong?

```
Train on 6 samples, validate on 2 samples
Epoch 1/10
6/6 [==============================] - 11s 2s/step - loss: 23.4232 - iou_score: 0.8338 - val_loss: 270.5751 - val_iou_score: 0.2407
Epoch 2/10
6/6 [==============================] - 0s 52ms/step - loss: -10.5406 - iou_score: 1.1558 - val_loss: 15.3342 - val_iou_score: 1.3462
Epoch 3/10
6/6 [==============================] - 0s 51ms/step - loss: -41.8985 - iou_score: 1.6151 - val_loss: -61.7260 - val_iou_score: 2.2258
Epoch 4/10
6/6 [==============================] - 0s 54ms/step - loss: -75.2752 - iou_score: 2.2215 - val_loss: -55.5733 - val_iou_score: 2.1647
Epoch 5/10
6/6 [==============================] - 0s 53ms/step - loss: -106.0778 - iou_score: 3.0294 - val_loss: -94.5141 - val_iou_score: 3.3411
Epoch 6/10
6/6 [==============================] - 0s 52ms/step - loss: -144.1045 - iou_score: 3.9788 - val_loss: -78.0447 - val_iou_score: 2.5654
Epoch 7/10
6/6 [==============================] - 0s 51ms/step - loss: -172.3163 - iou_score: 4.8828 - val_loss: -97.2337 - val_iou_score: 2.8799
Epoch 8/10
6/6 [==============================] - 0s 52ms/step - loss: -194.8336 - iou_score: 6.2343 - val_loss: -152.3081 - val_iou_score: 4.4320
Epoch 9/10
6/6 [==============================] - 0s 51ms/step - loss: -212.0709 - iou_score: 7.1287 - val_loss: -206.3270 - val_iou_score: 5.0225
Epoch 10/10
6/6 [==============================] - 0s 54ms/step - loss: -227.2620 - iou_score: 7.6453 - val_loss: -256.2039 - val_iou_score: 5.6111
```
Segmentation for a single class.

Dataset:
X:
![akvjocfxbvxldnuj](https://user-images.githubusercontent.com/5787026/57248886-d75fa480-7043-11e9-98c8-b779ca405323.png)

Y:
![akvjocfxbvxldnuj](https://user-images.githubusercontent.com/5787026/57248902-e0507600-7043-11e9-9bec-da3912fad707.png)

Code:
[unet_train.txt](https://github.com/qubvel/segmentation_models/files/3149491/unet_train.txt)

**Comment by qubvel** [comment]
Hi @fuzzyBatman  
make sure your masks are binary images {0, 1} (not {0, 255}).

**Comment by fuzzyBatman** [comment]
Legend. Thanks mate.
Is this the case for any Unet framework?

**Comment by qubvel** [comment]
Usually, output of the network is activated with softmax/sigmoid function, so it is in range 0..1. That is why your target should be in the same range, however it depends on loss function (mask normalisation can be done there).

**Comment by bfialkoff** [comment]
Is this the case for the input images as well?

**Comment by fuzzyBatman** [comment]
@bfialkoff No. I remember changing only the mask images.

**Comment by bfialkoff** [comment]
That's interesting. I got it to work by normalizing the input images as
well. Otherwise the negative loss persisted.
